packages/hub/src/lib/commit.ts

```diff
@@ -459,6 +464,35 @@ export async function* commitIter(params: CommitParams): AsyncGenerator<CommitPr
 
 		yield { event: "phase", phase: "committing" };
 
+		const allOpsToCommit = [...deleteOps, ...fileOpsToCommit];
+		if (allOpsToCommit.length === 0) {
+			const res: Response = await (params.fetch ?? fetch)(
+				`${params?.hubUrl || HUB_URL}/api/${repoId.type}s/${repoId.name}`,
+				{
+					headers: {
+						accept: "application/json",
+						...(params.credentials && { Authorization: `Bearer ${params.credentials.accessToken}` }),
+					},
+				}
+			);
+
+			if (!res.ok) {
+				throw await createApiError(res);
+			}
+
+			const { sha } = await res.json();
+
+			return {
+				commit: {
+					oid: sha,
+					url: `${params?.hubUrl || HUB_URL}${repoId.type !== "model" ? `/${repoId.type}s` : ""}/${
+						repoId.name
+					}/commit/${sha}`,
+				},
+				hookOutput: "Nothing to commit",
+			};
```

Comment:
I didn't want to do breaking changes by returning `null` in case we have nothing to commit, so returning this, similar to what @Wauplin did 🤷

Reply:
Hope it doesn't harm us in the future :smile: